### PR TITLE
Update package model, Create application services

### PR DESCRIPTION
**Pull request by smncrm**

### CODE REVIEW

File: src/main/java/mops/foren/applicationservices/PostService.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    /**
+     * Method to add a post.
+     *
+     * @param post     The post to add
+     * @param user     The user that wants to create the post
+     * @param threadId The threadId the post belongs to
+     */
+    public void addPost(Post post, User user, Long threadId) {
```
Comment (MarcoSteinke) [inline src/main/java/mops/foren/applicationservices/PostService.java]: Look into branch "permissions-and-new-domain-model". I wrapped the Id's so the Method checkPermission will know if the id is a ThreadId or ForumId.

File: src/main/java/mops/foren/applicationservices/ThreadService.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    /**
+     * Method to add a thread.
+     *
+     * @param thread  The thread to add
+     * @param user    The user that wants to create the thread
+     * @param topicId The topicId the thread belongs to
+     */
+    public void addThread(Thread thread, User user, Long topicId) {
```
Comment (MarcoSteinke) [inline src/main/java/mops/foren/applicationservices/ThreadService.java]: Look into branch "permissions-and-new-domain-model". I wrapped the Id's so the Method checkPermission will know if the id is a ThreadId or ForumId.

File: src/main/java/mops/foren/applicationservices/TopicService.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    /**
+     * Method to add a topic.
+     *
+     * @param topic   The thread to add
+     * @param user    The user that wants to create the topic
+     * @param forumId The forumId the topic belongs to
+     */
+    public void addTopic(Topic topic, User user, Long forumId) {
```
Comment (MarcoSteinke) [inline src/main/java/mops/foren/applicationservices/TopicService.java]: Look into branch "permissions-and-new-domain-model". I wrapped the Id's so the Method checkPermission will know if the id is a ThreadId or ForumId.

File: src/main/java/mops/foren/domain/model/User.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package mops.foren.domain.model;
+
+import javax.validation.constraints.Email;
+import java.util.HashMap;
+import java.util.List;
+
+public class User {
+    private String name;
+    private Email email;
+    private HashMap<Long, Permission> permissons;
```
Comment (MarcoSteinke) [inline src/main/java/mops/foren/domain/model/User.java]: In the branch "permissions-and-new-domain-model" I got to implement a permission-system which suits this storage of permissions inside of a HashMap. The next step would be to implement a PermissionService, which checks the user-permission on security-related events.

**Review by MarcoSteinke — CHANGES_REQUESTED** [review]
We will talk about the changes and maybe perform them tomorrow

File: src/main/java/mops/foren/domain/model/Forum.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package mops.foren.domain.model;
+
+public class Forum {
+    private ForumId id;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/Forum.java]: make this id variable final please

File: src/main/java/mops/foren/domain/model/ForumId.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package mops.foren.domain.model;
+
+public class ForumId implements Id {
+
+    private Long id;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/ForumId.java]: make this variable final please

File: src/main/java/mops/foren/domain/model/Post.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package mops.foren.domain.model;
+
+import lombok.RequiredArgsConstructor;
+
+import java.sql.Date;
+
+@RequiredArgsConstructor
+public class Post {
+    private PostId id;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/Post.java]: final please

File: src/main/java/mops/foren/domain/model/Post.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+    private ThreadId threadId;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/Post.java]: final please

File: src/main/java/mops/foren/domain/model/PostId.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package mops.foren.domain.model;
+
+public class PostId implements Id {
+
+    private Long id;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/PostId.java]: final please

File: src/main/java/mops/foren/domain/model/Thread.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package mops.foren.domain.model;
+
+public class Thread {
+
+    private ThreadId id;
+
+    private TopicId topicId;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/Thread.java]: final please

File: src/main/java/mops/foren/domain/model/ThreadId.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
+package mops.foren.domain.model;
+
+public class ThreadId implements Id {
+
+    private Long id;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/ThreadId.java]: final please

File: src/main/java/mops/foren/domain/model/Topic.java [hunk 12]
```diff
@@ -0,0 +1,11 @@
+package mops.foren.domain.model;
+
+public class Topic {
+    private TopicId id;
+
+    private ForumId forumId;
```
Comment (niruc100) [inline src/main/java/mops/foren/domain/model/Topic.java]: final please

**Review by mgtgecko — APPROVED** [review]
Läuft.

**Review by mgtgecko — APPROVED** [review]

**Review by MarcoSteinke — APPROVED** [review]

**Pull request merged**